python/pyspark/sql/connect/dataframe.py

```diff
@@ -201,6 +202,34 @@ def sort(self, *cols: "ColumnOrString") -> "DataFrame":
         """Sort by a specific column"""
         return DataFrame.withPlan(plan.Sort(self._plan, *cols), session=self._session)
 
+    def sample(
+        self,
+        withReplacement: Optional[Union[float, bool]] = None,
+        fraction: Optional[Union[int, float]] = None,
+        seed: Optional[int] = None,
+    ) -> "DataFrame":
+        from pyspark.sql import DataFrame as PySparkDataFrame
+
+        (withReplacement, fraction, seed) = PySparkDataFrame._prepare_augments_for_sample(
+            withReplacement, fraction, seed
+        )
+        if withReplacement is None:
+            withReplacement = False
```

Comment:
The default bool value for proto is `False` so this is probably not needed.

Reply:
oh The Plan definition is not `Optional` for `withReplacement`. In this case probably set it as `False` makes sense.

```
class Sample(LogicalPlan):
    def __init__(
        self,
        child: Optional["LogicalPlan"],
        lower_bound: float,
        upper_bound: float,
        with_replacement: bool,
        seed: int,
    ) -> None:
```

Reply:
yeah, let me make this change